src/pages/index.js

```diff
@@ -82,12 +131,37 @@ function getCardElement(data) {
 
   const cardLikeBtnEl = cardElement.querySelector(".card__like-button");
   cardLikeBtnEl.addEventListener("click", () => {
-    cardLikeBtnEl.classList.toggle("card__like-button_active");
+    const isLiked = cardLikeBtnEl.classList.contains(
+      "card__like-button_active"
+    );
+    const toggleLike = isLiked
+      ? api.removeLike.bind(api)
+      : api.addLike.bind(api);
+    toggleLike(data._id)
+      .then((updatedCard) => {
+        cardLikeBtnEl.classList.toggle(
+          "card__like-button_active",
+          updatedCard.isLiked
+        );
+      })
+      .catch(console.error);
   });
 
   const cardDeleteBtnEl = cardElement.querySelector(".card__delete-button");
   cardDeleteBtnEl.addEventListener("click", () => {
-    cardElement.remove();
+    selectedCard = cardElement;
+    selectedCardId = data._id;
+    openModal(deleteModal);
+  });
+
+  deleteCancelButton.addEventListener("click", () => {
+    selectedCard = null;
+    selectedCardId = null;
+    closeModal(deleteModal);
+  });
+
+  deleteCloseButton.addEventListener("click", function () {
+    closeModal(deleteModal);
```

Comment:
The `deleteCancelButton` and `deleteCloseButton` are the buttons of the `delete modal`. They should be handled only **once** outside the `getCardElement()` function in line 188